app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -266,6 +266,15 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
 
         requireComponents.core.engine.speculativeCreateSession(isPrivate)
 
+        // Handle the scenario in which the user selects another search engine before starting a search.
```

Comment:
nit: this logic could be moved to a func which would remove the need for the suppression and potentially self document

Reply:
Thank you!
Extracted the code to a separate method which also ensures easier testing. Would prefer to keep the current comment as context for why we call that method.

Reply:
Understood 👍 Appreciate the added tests